src/mainboard/sunxi/nezha/main/src/main.rs

```diff
@@ -21,53 +21,47 @@ use spin;
 
 mod sbi_platform;
 
-const MEM: usize = 0x4000_0000;
+const RAM_BASE: usize = 0x4000_0000;
 
 // see ../fixed-dtfs.dts
-// const PAYLOAD_OFFSET: usize = 0x2_0000;
-const PAYLOAD_SIZE: usize = 0x20_0000; // 2 MB
-const PAYLOAD_ADDR: usize = MEM + 0x20_0000;
+// target location for decompressed image
+const PAYLOAD_OFFSET: usize = 0x20_0000;
+const PAYLOAD_ADDR: usize = RAM_BASE + PAYLOAD_OFFSET;
+const PAYLOAD_SIZE: usize = 0x0200_0000; // 32 MB
 
-// compressed image
-const LINUXBOOT_TMP_OFFSET: usize = 0x0400_0000;
-const LINUXBOOT_TMP_ADDR: usize = MEM + LINUXBOOT_TMP_OFFSET;
+const DTB_ADDR: usize = PAYLOAD_ADDR + PAYLOAD_SIZE;
+const DTB_SIZE: usize = 0x0001_0000;
 
-// target location for decompressed image
-const LINUXBOOT_OFFSET: usize = 0x0020_0000;
-const LINUXBOOT_ADDR: usize = MEM + LINUXBOOT_OFFSET;
-const LINUXBOOT_SIZE: usize = 0x0200_0000;
-// DTB_OFFSET should be >=LINUXBOOT_OFFSET+LINUXBOOT_SIZE and match bt0
-// TODO: Should we just copy it to a higher address before decompressing Linux?
-const DTB_OFFSET: usize = LINUXBOOT_OFFSET + LINUXBOOT_SIZE;
-const DTB_ADDR: usize = MEM + DTB_OFFSET;
+const COMPRESSED_ADDR: usize = RAM_BASE + 0x0400_0000;
+const COMPRESSED_SIZE: usize = 0x00fe_0000;
 
 fn decompress_lb() {
```

Comment:
Note that this doesn't just decompress; it does a lot of checking and validation, too.